sentry_kube/cli/pg/create_user.py

```diff
@@ -124,35 +127,46 @@ def upload_plaintext_to_k8s_secret(
     users: dict[str, dict[str, str]],
     secret_name: str
 ) -> None:
-    # TODO: create if not exists
-    # api.create_namespaced_secret(namespace, body)
+    try:
+        secret = api.read_namespaced_secret(namespace="default", name=secret_name)
+    except ApiException as exc:
+        if exc.status == 404:
+            print(f"Secret `default/{secret_name}` does not exist, creating...")
+            body = V1Secret()
+            body.metadata = {"name": secret_name}
+            body.data = {}
+            api.create_namespaced_secret(namespace="default", body=body)
+            secret = api.read_namespaced_secret(namespace="default", name=secret_name)
```

Comment:
Is it possible to write a test case for this scenario i.e. create the secret if its missing.

Reply:
An example of unit test that extensively mocks the k8s api 
https://github.com/getsentry/devinfra-deployment-service/blob/main/gocd_agent/scripts/tests/k8s/k8s-deploy_test.py